flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializationTest.java

```diff
@@ -105,11 +119,235 @@ public void testHandleMixedLargeRecords() throws Exception {
 		testSerializationRoundTrip(originalRecords, segmentSize);
 	}
 
+	/**
+	 * Non-spanning, deserialization reads one byte too many and succeeds - failure report comes
+	 * from an additional check in {@link SpillingAdaptiveSpanningRecordDeserializer}.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning1() throws Exception {
```

Comment:
Why are you testing for a behaviour that actually is not the one that we would like to have? If this doesn't fail, then this test has no value and it's just an overhead for running/maintaining.

Reply:
Since the user can write custom serializers, we need to test that our code handles them as expected. This is what the test does. Or maybe I'm not getting your argument...

Reply:
But this test doesn't test for anything. Is our contract: if "deserialization reads one byte too many" the deserialisation must "succeed"? If this behaviour changes, and after refactoring we somehow throw an exception here would it be a bad thing? It looks to me like a no-op test that if it fails, it can be safely removed or changed to actually expecting the exception.

If at all, this should be tested in the `SpillingAdaptiveSpanningRecordDeserializer`'s test, where this "additional check" is kicking in.